### PR TITLE
Release v1.4.0

### PR DESCRIPTION
**Pull request by msm-code**
# Version 1.4

### Breaking changes

**[Breaking change]** mquery now users typedconfig library instead of the previous config.py file.

 - if you deployed mquery using docker (configurable by environment variables)
   then no action is required and this is backwards-compatible for you
 - if you deployed mquery natively using the default configuration, no action is required
 - finally, if you deploy mquery natively and changed the default config.py, you will have to create a mquery.ini
   file with your config. The format is very simple. Example of a complete config file (there are only 4 possible
   configuration keys supported currently. All are optional):

```
[redis]
host=localhost
port=6379

[mquery]
backend=tcp://localhost:9281
plugins=
```

### New features

- It's now possible to limit the number of yara-scanned files (#339)
- It's now possible to disallow running slow queries (#315, #312)
- Added a configurable /about page, to describe your instance (#341)
- Daemon now has a --scale flag, to automatically fork into mutliple processes (#298)
- More flexible user roles (#350, #314)

### Documentation

- Mquery component documentation (#334)
- Yara support documentation (#333)
- S3 support documentation (#327)

### UI Improvements

- Progress bar now shows more information (#345)
- Counter race condition fixed (#348)
- Bootstrap update and following fixes (#346,

### Improvements

- A big backend improvement - jobs are now scheduled with a rq framework (#317)
- Exceptions thrown during filtering with plugins are now handled correctly (#317)
- Login is now faster - there are no unnecessary redirects (#322)

### Bugfixes

- /about route fixed (#343)
- Indexing script won't skip the last few files anymore (#328)
- Actually raise errors from the API (#311)
- Fix multi-agent job completion (#282)

### Others

- Dockerignore and Gitignore updated (#344)
- Some obsolete features removed from the codebase (#330, #313, #306)

### CODE REVIEW

File: RELEASES.md [hunk 1]
```diff
@@ -1,3 +1,65 @@
...
+port=6379
+
+[mquery]
+backend=tcp://localhost:9281
+plugins=
+```
+
+### New features
+
+- It's now possible to limit the number of yara-scanned files (#339)
```
Comment (nazywam) [inline RELEASES.md:26]:
```suggestion
- It's now possible to limit the number of YARA-scanned files (#339)
```

File: RELEASES.md [hunk 2]
```diff
@@ -1,3 +1,65 @@
...
+- It's not possible to disallow running slow queries (#315, #312)
```
Comment (nazywam) [inline RELEASES.md:27]:
```suggestion
- It's now possible to disallow running slow queries (#315, #312)
```
seems like a regression to me

File: RELEASES.md [hunk 3]
```diff
@@ -1,3 +1,65 @@
...
+- Added a configurable /about page, to describe your instance (#341)
+- Daemon now has a --scale flag, to automatically fork into mutliple processes (#298)
+- More flexible user roles (#350, #314)
+
+### Documentation
+
+- Mquery component documentation (#334)
+- Yara support documentation (#333)
```
Comment (nazywam) [inline RELEASES.md:35]:
```suggestion
- YARA support documentation (#333)
```

File: RELEASES.md [hunk 4]
```diff
@@ -1,3 +1,65 @@
...
+
+### UI Improvements
+
+- Progress bar now shows more information (#345)
+- Counter race condition fixed (#348)
+- Bootstrap update and following fixes (#346, 
+
+### Improvements
+
+- A big backend improvement - jobs are now scheduled with a rq framework (#317)
```
Comment (nazywam) [inline RELEASES.md:46]:
```suggestion
- A big backend improvement - jobs are now scheduled using the RQ framework (#317)
```

File: src/app.py [hunk 5]
```diff
@@ -105,23 +105,31 @@ async def current_user(authorization: Optional[str] = Header(None)) -> User:
...
-        return User(None)
+    # 401 error object, that will force the user to re-authenticate.
+    unauthorized = HTTPException(
+        status_code=401,
+        detail="Invalid token, please re-authenticate.",
+    )
+
+    # Be nice for the user, even when they send us an invalid header.
+    token_parts = authorization.split()
+    if len(token_parts) != 2 or token_parts[0] != "Bearer":
```
Comment (nazywam) [inline src/app.py:116]: I don't think this should be here? I've seen this change in the past already

**Comment by nazywam** [comment]
The `cp` commit was probably added by mistake, other than that - looks ok

File: RELEASES.md (hunk 2 above)
Comment (msm-code) [reply RELEASES.md:27]: :snake:

**Review by nazywam — APPROVED** [review]

**Pull request merged**